services/spar/src/Spar/API.hs

```diff
@@ -441,23 +430,37 @@ idpDelete zusr idpid (fromMaybe False -> purge) = withDebugLog "idpDelete" (cons
     IdPRawMetadataStore.delete idpid
   pure NoContent
   where
+    assertEmptyOrPurge :: TeamId -> Cas.Page (SAML.UserRef, UserId) -> Sem r ()
+    assertEmptyOrPurge team page = do
+      forM_ (Cas.result page) $ \(uref, uid) -> do
+        mAccount <- BrigAccess.getAccount NoPendingInvitations uid
+        let mUserTeam = userTeam . accountUser =<< mAccount
+        when (mUserTeam == Just team) $ do
+          if purge
+            then do
+              BrigAccess.delete uid
+              SAMLUserStore.delete uid uref
+            else do
+              throwSparSem SparIdPHasBoundUsers
+      when (Cas.hasMore page) $
+        SAMLUserStore.nextPage page >>= assertEmptyOrPurge team
+
     updateOldIssuers :: IdP -> Sem r ()
     updateOldIssuers _ = pure ()
     -- we *could* update @idp ^. SAML.idpExtraInfo . wiReplacedBy@ to not keep the idp about
     -- to be deleted in its old issuers list, but it's tricky to avoid race conditions, and
     -- there is little to be gained here: we only use old issuers to find users that have not
     -- been migrated yet, and if an old user points to a deleted idp, it just means that we
-    -- won't find any users to migrate.  still, doesn't hurt much to look either.  so we
+    -- won't find any users to migrate.  still, doesn't hurt mucht to look either.  so we
```

Comment:
?

Reply:
oops.  when i re-read this, i mixed up red and green, and thought it was *fixing* a typo.  oh well.